recipe/meta.yaml

```diff
@@ -38,6 +41,11 @@ requirements:
   host:
     - r-base
     - libcurl                        # [not win]
+    # Test below (connecting to rest.kegg.jp) triggers upstream issue described in
+    # https://github.com/curl/curl/issues/13389 and https://github.com/openssl/openssl/issues/11378
+    # Building against libcurl built against openssl<3.2 allows to circumvent that issue if needed.
+    # TODO: Remove explicit dependency when workaround is not needed anymore.
+    - openssl <3.2.0a0               # [not win]
```

Comment:
@bernt-matthias, you added the download test for `http://rest.kegg.jp/conv/ncbi-geneid/dme` in gh-21.
I was going to just remove that test because it is an upstream issue, really.
But for now I only added this workaround here with building against an older `libcurl` (as of now, it picks `8.4` instead of the current `8.8`) which allows `r-curl` to be installed alongside `openssl<3.2` to be able to avoid that upstream issue if needed.

N.B.: All of that should be fine since we dynamically link against `libcurl` such that normally users will get the latest `libcurl`/`openssl` versions installed.
It also does *not* collide with our pinnings since we pin against `libcurl=8` and `openssl=3`, i.e., the any of those `8.x`/`3.x` versions is okay to link against as per our pinning.